.github/workflows/go-ci.yml

```diff
@@ -149,3 +149,37 @@ jobs:
     uses: ./.github/workflows/integration-tests.yml
     with:
       go-version: ${{ inputs.go-version }}
+
+  e2e_test:
+    name: E2E tests
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        test:
+          - TestE2EMsgPayForBlob  # Expand as new E2E tests are added
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version-file: go.mod
+
+      - name: Generate image tag
+        id: tag
+        run: echo "value=$(echo $GITHUB_SHA | cut -c1-8)" >> $GITHUB_OUTPUT
+
+      - name: Build Docker image locally
+        run: docker build -t "celestia-node:${{ steps.tag.outputs.value }}" .
```

Comment:
for now I am intentionally not pushing to ghcr.io/celestia-node as the existing re-usable docker workflow does that, and don't want to clash with anything there.

This adds a bit of time to each run, and can be re-considered later if it is a problem.